nym-registration-client/src/builder/config.rs

```diff
@@ -179,6 +181,59 @@ fn mixnet_debug_config(mixnet_client_config: &MixnetClientConfig) -> DebugConfig
     if let Some(min_gateway_performance) = mixnet_client_config.min_gateway_performance {
         debug_config.topology.minimum_gateway_performance = min_gateway_performance;
     }
+    if let Some(avg_packet_ms) = mixnet_client_config.average_packet_delay {
+        debug_config.traffic.average_packet_delay = avg_packet_ms;
+        debug_config.acknowledgements.average_ack_delay = avg_packet_ms;
+    }
+    // Decide whether the real-traffic Poisson process should be disabled.
+    //
+    // We disable it if:
+    // 1. The user explicitly requested it via `disable_real_traffic_poisson_process`, OR
+    // 2. The user set `message_sending_average_delay` to 0ms (special value meaning "disable")
+    let mut disable_real_traffic_poisson =
+        mixnet_client_config.disable_real_traffic_poisson_process;
+
+    if let Some(delay) = mixnet_client_config.message_sending_average_delay
+        && delay.is_zero()
+    {
+        disable_real_traffic_poisson = true;
+    }
+
+    debug_config
+        .traffic
+        .disable_main_poisson_packet_distribution = disable_real_traffic_poisson;
+
+    // Only apply a custom message sending delay if the Poisson process is enabled.
+    // If Poisson is disabled, any delay value is ignored.
+    if !disable_real_traffic_poisson
```

Comment:
nit: It's correct, but we don't really need that extra condition on `disable_real_traffic_poisson` imo. If it's disabled, it will get ignored anyway